src/server/replica.cc

```diff
@@ -183,7 +188,7 @@ std::error_code Replica::TakeOver(std::string_view timeout, bool save_flag) {
   return ec;
 }
 
-void Replica::MainReplicationFb() {
+void Replica::MainReplicationFb(std::optional<LastMasterSyncData> data) {
```

Comment:
`LastMasterSyncData` and `data` are too generic.
`LastMasterSyncVector` and `sync_vec` could be better names

Reply:
LastMasterSyncData is a struct that contains the master id and a vector of lsns.
IMO LastMasterSyncVector is not a better name for this struct. I will rename data though